presto-spi/src/main/java/com/facebook/presto/spi/predicate/AndExpression.java

```diff
@@ -87,6 +87,18 @@ public String getName()
     }
 
     @Override
+    public boolean isNone()
+    {
+        return leftExpression.isNone() || rightExpression.isNone();
```

Comment:
We can assume an `AndExpression` to be should not have any values(isNone() represents that) if any of the operands have no values in their domain. So we cant use && here.